osbuild/testutil/dnf4.py

```diff
@@ -0,0 +1,36 @@
+import tempfile
+from typing import List, Optional
+
+import dnf
+
+
+def depsolve_pkgset(
+    repo_paths: List[str],
+    pkg_include: List[str],
+    pkg_exclude: Optional[List[str]] = None
```

Comment:
I'm not super sure on libdnf but is there a difference between an empty list and `None` on that level?

For this function signature I would opt to not make `pkg_exclude` optional; depsolving always needs to receive includes and excludes.

Reply:
This is a helper function just for tests. The code uses it only to depsolve a transaction which actually never uses any excludes, so that it can be then converted to the internal SBOM model.

[`install_specs()` defaults to `None` for optional arguments](https://dnf.readthedocs.io/en/latest/api_base.html#dnf.Base.install_specs), therefore I used that. I didn't test the behavior with an empty list.